src/Bridges/Constraint/bridges/semi_to_binary.jl

```diff
@@ -75,6 +75,19 @@ function bridge_constraint(
     f::MOI.VariableIndex,
     s::S,
 ) where {T<:Real,S<:Union{MOI.Semicontinuous{T},MOI.Semiinteger{T}}}
+    F = MOI.VariableIndex
+    if MOI.is_valid(model, MOI.ConstraintIndex{F,MOI.GreaterThan{T}}(f.value))
+        throw(MOI.LowerBoundAlreadySet{MOI.GreaterThan{T},S}(f))
+    end
+    if MOI.is_valid(model, MOI.ConstraintIndex{F,MOI.LessThan{T}}(f.value))
+        throw(MOI.UpperBoundAlreadySet{MOI.LessThan{T},S}(f))
+    end
+    if MOI.is_valid(model, MOI.ConstraintIndex{F,MOI.EqualTo{T}}(f.value))
+        throw(MOI.LowerBoundAlreadySet{MOI.EqualTo{T},S}(f))
+    end
+    if MOI.is_valid(model, MOI.ConstraintIndex{F,MOI.Interval{T}}(f.value))
```

Comment:
If I understand correctly, this is because `Semicontinuous` sets bounds so it should be incompatible with existing bounds. In fact, if these bounds are removed before `optimize!` we shouldn't throw so we should really use `final_touch` here too. I guess it's another argument for removing these errors as suggested in https://github.com/jump-dev/MathOptInterface.jl/pull/1879#issuecomment-1158621338

Reply:
Maybe put that in another PR if it's not needed to pass the tests so that we can merge the rest